lib/stencil-init.js

```diff
@@ -37,15 +37,30 @@ class StencilInit {
      * @returns {Promise<void>}
      */
     async run (dotStencilFilePath, cliOptions = {}) {
+
+        const cliConfig = this.getCliConfig(cliOptions);
         const oldStencilConfig = this.readStencilConfig(dotStencilFilePath);
-        const defaultAnswers = this.getDefaultAnswers(oldStencilConfig, cliOptions);
-        const answers = await this.askQuestions(defaultAnswers);
-        const updatedStencilConfig = this.applyAnswers(oldStencilConfig, answers);
+        const defaultAnswers = this.getDefaultAnswers(oldStencilConfig);
+        const questions = this.getQuestions(defaultAnswers, cliConfig);
+        const answers = questions ? await this.askQuestions(questions) : {};
+        const updatedStencilConfig = this.applyAnswers(oldStencilConfig, answers, cliConfig);
         this.saveStencilConfig(updatedStencilConfig, dotStencilFilePath);
 
         this.logger.log('You are now ready to go! To start developing, run $ ' + 'stencil start'.cyan);
     }
 
+    /**
+     * @param {{port: (number), url: (string), token: (string)}} cliOptions
+     * @returns {{port: (number), normalStoreUrl: (string), accessToken: (string)}}
+     */
+    getCliConfig (cliOptions) {
+        return {
```

Comment:
Let's better accept cliOptions in the needed format from the very beginning, so we don’t need to update their shape here. Kinda:

*/bin/stencil-init.js:*
```
...
new StencilInit().run(dotStencilFilePath, {
    normalStoreUrl: cliOptions.url,
    accessToken: cliOptions.token,
    port: cliOptions.port,
});
...
```